frontend/src/metabase/visualizations/lib/LineAreaBarPostRenderer.js

```diff
@@ -0,0 +1,323 @@
+const MARGIN_BOTTOM_MIN     = 10;
+const MARGIN_HORIZONTAL_MIN = 20;
+
+// extra padding for axis
+const X_AXIS_PADDING = 0;
+const Y_AXIS_PADDING = 8;
+
+function adjustMargin(chart, margin, direction, padding, axisSelector, labelSelector) {
+    const axis      = chart.select(axisSelector).node();
+    const label     = chart.select(labelSelector).node();
+    const axisSize  = axis  ? axis.getBoundingClientRect()[direction] + 10 : 0;
+    const labelSize = label ? label.getBoundingClientRect()[direction] + 5 : 0;
+    chart.margins()[margin] = axisSize + labelSize + padding;
+}
+
+function computeMinHorizontalMargins(chart) {
+    let min = { left: 0, right: 0 };
+    const ticks = chart.selectAll(".axis.x .tick text")[0];
+    if (ticks.length > 0) {
+        const chartRect = chart.select("svg").node().getBoundingClientRect();
+        min.left = chart.margins().left - (ticks[0].getBoundingClientRect().left - chartRect.left);
+        min.right = chart.margins().right - (chartRect.right - ticks[ticks.length - 1].getBoundingClientRect().right);
+    }
+    return min;
+}
+
+function beforeRenderFixMargins(chart, settings) {
+    // run before adjusting margins
+    const mins = computeMinHorizontalMargins()
+
+    // adjust the margins to fit the X and Y axis tick and label sizes, if enabled
+    adjustMargin(chart, "bottom", "height", X_AXIS_PADDING, ".axis.x",  ".x-axis-label",          settings["graph.x_axis.labels_enabled"]);
+    adjustMargin(chart, "left",   "width",  Y_AXIS_PADDING, ".axis.y",  ".y-axis-label.y-label",  settings["graph.y_axis.labels_enabled"]);
+    adjustMargin(chart, "right",  "width",  Y_AXIS_PADDING, ".axis.yr", ".y-axis-label.yr-label", settings["graph.y_axis.labels_enabled"]);
+
+    // set margins to the max of the various mins
+    chart.margins().top    = Math.max(MARGIN_TOP_MIN,        chart.margins().top);
+    chart.margins().left   = Math.max(MARGIN_HORIZONTAL_MIN, chart.margins().left,  mins.left);
+    chart.margins().right  = Math.max(MARGIN_HORIZONTAL_MIN, chart.margins().right, mins.right);
+    chart.margins().bottom = Math.max(MARGIN_BOTTOM_MIN,     chart.margins().bottom);
+}
+
+// collection of function calls that get made *before* we tell the Chart to render
+function beforeRender(chart, settings) {
+    beforeRenderHideDisabledAxesAndLabels(chart, settings);
+    beforeRenderFixMargins(chart, settings);
+}
+
+// +-------------------------------------------------------------------------------------------------------------------+
+// |                                              PUTTING IT ALL TOGETHER                                              |
+// +-------------------------------------------------------------------------------------------------------------------+
+
+/// once chart has rendered and we can access the SVG, do customizations to axis labels / etc that you can't do through dc.js
+export default function lineAndBarOnRender(chart, settings, onGoalHover, isSplitAxis, isStacked) {
```

Comment:
This is a weird function name ... what exactly is it supposed to be doing?

Reply:
I didn't change the name of this function, just moved it to a different file. My understanding is that it is the `onRender` function that gets called for line *and* bar charts (as well as area and scatter charts)